pkg/app/application_test.go

```diff
@@ -34,6 +34,7 @@ import (
 func setupTest(t *testing.T) loader.Loader {
 	kustomizationContent := []byte(`
 namePrefix: foo-
+namespace: foo
```

Comment:
something other than the string used as the nameprefix please - e.g.  _banana_.  To lower the cognition needed to read the test.

Reply:
also, at Line 56 or so, please add a namespace.yaml to the base, as a sibling file to the deployment.yaml.

The goal is to get this test to document what happens when a _nameprefix_ and _namespace_ are specified in the same kustomization.

It would be bad to have the _nameprefix_ be applied to the _namespace_ specified in the same kustomization.  A namespace (like a node) is a property of the cluster, its not an object in the cluster (like a deployment or service).  Nameprefixes should only apply to the latter (objects). 

I can see an argument that the nameprefix should be applied to namespaces defined lower in the kustomization hierachy, but we don't have to decide that in this PR.  It can happen in a later PR.

Reply:
Ok updated